pkg/operator/ceph/object/objectstore.go

```diff
@@ -849,6 +849,10 @@ func GetObjectBucketProvisioner(data map[string]string, namespace string) string
 	if strings.EqualFold(obcWatchOnNamespace, "true") {
 		provName = fmt.Sprintf("%s.%s", namespace, bucketProvisionerName)
 	}
+	obcProvisionerNamePrefix := k8sutil.GetValue(data, "ROOK_OBC_PROVISIONER_NAME_PREFIX", "")
+	if obcProvisionerNamePrefix != "" {
+		provName = fmt.Sprintf("%s.%s", obcProvisionerNamePrefix, bucketProvisionerName)
```

Comment:
what happens if the user provides a prefix with extra `.`?

Reply:
will handle and makes the name should contain small alphabets + digits + special `-` ?

Reply:
Sounds good or we could add a check and return error if the prefix is ending with `.` that's what i did for csi driver.
https://github.com/rook/rook/blob/64cf8f22e5a633d514b3034511b97a5e7c893e24/pkg/operator/ceph/csi/spec.go#L322-L327